aiohttp/resolver.py

```diff
@@ -26,6 +27,15 @@ def __init__(self, loop=None):
         self._loop = loop
 
     async def resolve(self, host, port=0, family=socket.AF_INET):
+        if family == 0:
+            hosts = await self._resolve(host, port, family=socket.AF_INET)
+            hosts.extend(await self._resolve(
+                host, port, family=socket.AF_INET6))
```

Comment:
I have the feeling that both asynchronous calls can be run concurrently ....

Reply:
Just found that today [1], regarding the part that this code is trying to achieve the recommendation by the RFC is launch both asynchronously but always first the IPV6 to give it more priority.

In any case, most of the algorithm presented by the RFC is not followed by Aiohttp. At some point we might change the implementation to make it more aligned, I like some of the ideas.

[1] https://tools.ietf.org/html/rfc8305#page-4

Reply:
Thanks for the link.
I should find a time for a second attempt.